### PR TITLE
Exit bug for screenshot overlay

### LINKED ISSUE #22: Exit and save button not working from screenshot overlay
* **What Operating System?**

* **What version of Chrome?**

* **What version of Recorder's Assistant (GitHub version or Chrome Store version)?**

* **What steps will reproduce the issue?**
User trying to save and exit from the screenshot screen will experience nothing happening and even freezing of the extension sometimes
* **What would you expect to be the outcome?**
The exit and save button should work even from the screenshot overlay
* **Any additional details/screenshots to help people to fix any potential bugs?**

### PR DESCRIPTION
**Pull request by fatimamah**
---
name: Pull request template
about: GenderMag Recorders Assistant project

---

* **What does this implementation fix? Explain your changes.**	
	The save and exit button now works in the screenshot pop up/overlay and doesn't freeze if the user clicks "back." The issue was that it required the image url and name to let the user exit, but the name and url kept showing up as null so the program would freeze.

* **Does this close any open issue? (Give issue id from issue tracker)**

* **Include any related logs, error, output file etc.**

* **Did you test on Mac and Windows?**

* **Did you test the full GM process and verify that the code does not contribute any unexpected errors/bugs?**

* **If possible, provide a screenshot of the functionality.**

* **Any other information?**
Now, the program doesn't freeze and lets the user exit, but it downloads the file automatically without the user clicking anything and no image is downloaded because of the change (if uri or name is null, return). When it downloads, a pop up appears instead of it downloading to the bottom of the screen.

### CODE REVIEW

File: scripts/output.js [hunk 1]
```diff
@@ -233,7 +233,13 @@ function downloadCSV(csvContent, old) {
 	create_zip(csvContent, old);
 }
 
+/*
+*This function...
```
Comment (yuxiexiao) [inline scripts/output.js]: Can you finish the comments here to document what the downloadURI method does? You should mention that if the name or uri passed in are empty, we don't append the image object into the stored image list.

File: scripts/output.js [hunk 2]
```diff
@@ -233,7 +233,13 @@ function downloadCSV(csvContent, old) {
...
+*/
 function downloadURI(uri, name) {
+    if (uri === null || name ===null);{
```
Comment (yuxiexiao) [inline scripts/output.js]: super nit-pick, but can you add a space after the second '===' ?

**Review by yuxiexiao — CHANGES_REQUESTED** [review]

**Comment by anikuri** [comment]
Sorry for the messy commits...I'm struggling to figure out how to rebase, squash, merge, and amend commits.

File: scripts/output.js [hunk 3]
```diff
@@ -237,10 +237,9 @@ function downloadCSV(csvContent, old) {
 *This function...
 */
 function downloadURI(uri, name) {
-    if (uri === null || name ===null);{
-        return;
-    }
-	var safeName = name;
+    if (uri !== null) {
```
Comment (myradeng) [inline scripts/output.js]: I think we should maintain the functionality that checks if uri or name is null, and returns immediately if so. Even better, we should throw an error if we detect that either are null so the user knows what went wrong.

File: scripts/output.js [hunk 4]
```diff
@@ -237,10 +237,9 @@ function downloadCSV(csvContent, old) {
...
+        var safeName = name;
+    toString(uri);
```
Comment (myradeng) [inline scripts/output.js]: Just wondering, why do we need this toString(uri)?

**Comment by anikuri** [comment]
@emlabor Could you please check if this works on Mac?

File: scripts/output.js [hunk 5]
```diff
@@ -233,17 +233,29 @@ function downloadCSV(csvContent, old) {
...
-	var imgObj = {
-		name:safeName+".png",
-		uri: safeUri
-	};
-	console.log("imgobj", imgObj);
-	imgList.push(imgObj);
-
+    try {
+        //checks to see if the uri or name is null
+        if (uri === null || name === null) throw "The uri or name for your image is null.";
```
Comment (emlabor) [inline scripts/output.js:242]: Is the idea that if "Exit" is clicked this error will be logged? (For me, I can't get this error to show up)

**Review by emlabor — APPROVED** [review]
This such a nice feature 😄 And it does work on Mac.

File: scripts/output.js (hunk 5 above)
Comment (anikuri) [reply scripts/output.js:242]: It's just error checking to make sure the URI and name isn't null lol, it should only show the error when there is an error (which there shouldn't be)

Comment (emlabor) [reply scripts/output.js:242]: Lol okay

**Pull request merged**